jerry-core/ecma/builtin-objects/ecma-builtin-helpers.cpp

```diff
@@ -299,16 +299,14 @@ ecma_builtin_helper_array_index_normalize (ecma_number_t index, /**< index */
     }
     else
     {
-      const int32_t int_index = ecma_number_to_int32 (index);
-
-      if (int_index < 0)
+      if (ecma_number_is_negative (index))
       {
-        const uint32_t uint_index = (uint32_t) - int_index;
+        const uint32_t uint_index = ecma_number_to_uint32 (ecma_number_negate (index));
         norm_index = uint_index > length ? 0 : length - uint_index;
       }
       else
       {
-        norm_index = (uint32_t) int_index;
+        norm_index = ecma_number_to_uint32 (index);
```

Comment:
@dbatyai, 15.4.4.10.5 uses `ToInteger` that doesn't perform `modulo 2^32` operation during conversion. Maybe, this can lead to incorrect conversion in the cases, if `index` is more than 2^32.